third_party/move/move-vm/runtime/src/runtime.rs

```diff
@@ -34,9 +34,18 @@ use move_vm_types::{
 use std::{borrow::Borrow, collections::BTreeSet, sync::Arc};
 
 /// An instantiation of the MoveVM.
-#[derive(Clone)]
 pub(crate) struct VMRuntime {
     loader: Loader,
+    pub(crate) module_cache: Arc<ModuleCache>,
```

Comment:
It is now quite interesting that the Loader is still stateful even after the module cache being moved out. I feel like there is still some work we can do here to better manage these states, and their hierarchy.